### PR TITLE
Add commit checker job and scripts

### PR DESCRIPTION
**Pull request by ystefinko**
Separate job must run only on PR stage.
Job verify commit message and fails if
requirements are not met.
Requirements added into file:
 - scripts/misc/commit_message_recom.txt

Relates-To: OLPEDGE-1573

Signed-off-by: Yaroslav Stefinko <ext-yaroslav.stefinko@here.com>

### CODE REVIEW

File: scripts/misc/commit_message_recom.txt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- Bullet points are okay, too.
+
+- Typically a hyphen or asterisk is used for the bullet, preceded
+  by a single space, with blank lines in between, but conventions
+  vary here.
+
+All commits need to reference a ticket, again separated by a blank
+line from the commit message summary above:
+
+Resolves: ABC-111, #123
```
Comment (OleksiiZubko) [inline scripts/misc/commit_message_recom.txt:28]: It's required? What about CRs from external guys if it's required?

File: scripts/misc/commit_checker.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+relates_to=$(cat commit.log | grep 'Relates-To: ') || true
+resolves=$(cat commit.log | grep 'Resolves: ') || true
+see=$(cat commit.log | grep 'See also: ') || true
+
+echo "Reference like:  ${relates_to} ${resolves} ${see}  was found in commit message."
+
+if [[ -n ${relates_to} || -n ${resolves} || -n ${see} ]] ; then
+    echo "Commit message contains Jira link."
+else
+    echo "ERROR: Commit message does not contain ticket reference like Relates-To: or Resolves: or See also:  "
```
Comment (OleksiiZubko) [inline scripts/misc/commit_checker.sh:71]: not sure about that. I meant CRs from external guys.

File: scripts/misc/commit_message_recom.txt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+Relates-To: ABC-333, #321
+See also: ABC-432, #456, #789
+
+Signed-off-by: FirstName LastName <firstname.lastname@here.com>
```
Comment (OleksiiZubko) [inline scripts/misc/commit_message_recom.txt:32]: not sure about here.com. We're opensorce and any github user are able to create CR

File: scripts/misc/commit_message_recom.txt (hunk 1 above)
Comment (ystefinko) [reply scripts/misc/commit_message_recom.txt:28]: good point, comment was adjusted accordingly.

File: scripts/misc/commit_checker.sh (hunk 2 above)
Comment (ystefinko) [reply scripts/misc/commit_checker.sh:63]: good point, comment was adjusted accordingly.

File: scripts/misc/commit_message_recom.txt (hunk 3 above)
Comment (ystefinko) [reply scripts/misc/commit_message_recom.txt]: good point, email was changed accordingly.

**Pull request merged**